src/main/java/digital/slovensko/autogram/core/SignatureValidator.java

```diff
@@ -0,0 +1,153 @@
+package digital.slovensko.autogram.core;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import eu.europa.esig.dss.model.DSSDocument;
+import eu.europa.esig.dss.service.crl.OnlineCRLSource;
+import eu.europa.esig.dss.service.http.commons.CommonsDataLoader;
+import eu.europa.esig.dss.service.http.commons.FileCacheDataLoader;
+import eu.europa.esig.dss.service.http.commons.OCSPDataLoader;
+import eu.europa.esig.dss.service.ocsp.OnlineOCSPSource;
+import eu.europa.esig.dss.spi.tsl.TrustedListsCertificateSource;
+import eu.europa.esig.dss.spi.x509.CertificateSource;
+import eu.europa.esig.dss.spi.x509.KeyStoreCertificateSource;
+import eu.europa.esig.dss.spi.x509.aia.DefaultAIASource;
+import eu.europa.esig.dss.tsl.function.OfficialJournalSchemeInformationURI;
+import eu.europa.esig.dss.tsl.job.TLValidationJob;
+import eu.europa.esig.dss.tsl.source.LOTLSource;
+import eu.europa.esig.dss.tsl.sync.AcceptAllStrategy;
+import eu.europa.esig.dss.validation.CommonCertificateVerifier;
+import eu.europa.esig.dss.validation.SignedDocumentValidator;
+import eu.europa.esig.dss.validation.reports.CertificateReports;
+import eu.europa.esig.dss.validation.reports.Reports;
+
+public class SignatureValidator {
+    private static final String LOTL_URL = "https://ec.europa.eu/tools/lotl/eu-lotl.xml";
+    private static final String OJ_URL = "https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=uriserv:OJ.C_.2019.276.01.0001.01.ENG";
+    private CertificateSource journalCertificateSource;
+    private FileCacheDataLoader dataLoader;
+
+    public Reports validate(DSSDocument doc, SignedDocumentValidator docValidator) {
+        var trustedListCertificateSource = new TrustedListsCertificateSource();
+        processLOTL(trustedListCertificateSource);
+
+        var verifier = new CommonCertificateVerifier();
+        verifier.setTrustedCertSources(trustedListCertificateSource);
+        verifier.setAIASource(new DefaultAIASource(dataLoader));
+        verifier.setCrlSource(new OnlineCRLSource(dataLoader));
+        verifier.setOcspSource(new OnlineOCSPSource(new OCSPDataLoader()));
+
+        docValidator.setCertificateVerifier(verifier);
+        var r = docValidator.validateDocument();
+
+        System.out.println("Done");
+        saveResults(r);
+
+        return r;
+    }
+
+    private void processLOTL(TrustedListsCertificateSource trustedListCertificateSource) {
+        try {
+            journalCertificateSource = new KeyStoreCertificateSource(
+                new File("/home/turtle/slovensko_digital/autogram/src/main/resources/lotlKeyStore.p12"), "PKCS12", "dss-password");
```

Comment:
Toto je pomerne divná vec. List of Trusted Lists (napr pre EU) obyčajne nie je nijako trusted, keď to iba stiahneme z nejakej adresy. Preto existuje na internete 8 certifikátov tohto LOTL. Aj podľa DSS oficiálnej dokumentácie s examplami sa to má robiť tak, že do tohto keystoru človek uloží tých 8 ertifikátov, ktorými sa validuje samotný LOTL.

Tak som manuálne z oficiálnej stránky okopíroval tie certifikáty v textovej forme a nakoniec ich importol do tohoto keystoru. A takto to funguje. LOTL je teraz "well signed". Avšak, zdá sa mi divné, že keystore je chránený nejakým heslom, ktoré je public. Ale možno je tam to heslo iba preto, lebo to Java nevie bez hesla.